### DISCUSSION TITLE
Starting Quarkus 3

### DISCUSSION THREAD

**Discussion by giscus** [quarkusio/quarkus#29183] [Quarkus Blog/Website]
# Starting Quarkus 3

Quarkus: Supersonic Subatomic Java

https://quarkus.io/blog/road-to-quarkus-3/

**Comment by adampoplawski** [comment]
Hello
In Quarkus 3.0 there is a plan to upgrade Hibernate ORM to 6.0. I assume this only refers to Classic Hibernate not Reactive one?

**Reply by uPagge** [reply]
https://quarkus.io/blog/road-to-quarkus-3/

<img width="841" alt="screen" src="https://user-images.githubusercontent.com/40397740/201468351-08d91b1e-a4e8-469b-b549-5476b4ddf234.png">

**Reply by maxandersen** [reply]
Hibernate reactive would need to get updated to v6 to be part of quarkus 3.

**Reply by Sanne** [reply]
Right, we'll upgrade Hibernate ORM to 6.x (likely 6.2) and we're working on an Hibernate Reactive version based on it - will likely be Hibernate Reactive 2.0

**Comment by ArturSkowronski** [comment]
Hey Max,
thanks for heads up. Do you have any public info how you want to use io_uring and wha "move from Reactive Streams" means in practice?

Thanks in advance, you got me curious.

**Reply by franz1981** [reply]
Io_uring is still a pr on Vertx core and I would like to merge it asap.
Now is based on the Netty 4.1 incubator.
Netty 5 instead will make it a first citizen (and more performant) option but I am not aware yet what are the plans for Netty 5 adoption on Quarkus, meaning that I will backport to the incubator (I am a Netty committer) whatever is needed in order to get its benefit with Netty 4.1 that's currently used on Vertx (and quarkus, by consequence).

In term of effects it means that having the right kernel versions users will be benefit of more performant TCP and UDP stack, especially visible with many connections.
It means much less I/O threads will be required to achieve the current level of performance of Epoll and that means more CPU cycles available to the reactive stack (and Netty/Vertx encoders/decoders).
From an usage point of view it will be fully transparent to users, given that's just a transport implementation detail.

**Reply by jponge** [reply]
Mutiny is moving from the legacy reactive streams APIs (Java 6 era) to JDK Flow (since Java 9).

This is transparent for user code based on Mutiny.

Quarkus will offer a transition by supporting both types in places where it makes sense (RestEasy Reactive, Reactive Messaging, etc). In other place adapters will be used.

This will be documented soon.

**Reply by ArturSkowronski** [reply]
@jponge @framz1981 thanks! 

@jponge Which `legacy reactive streams APIs` you means? If I remember correctly, support for Reactive Streams was added in JDK 9.

Still, I'll probably just wait for docs here :) Great to see an ongoing progress in the Reactive support topic.

**Reply by jponge** [reply]
> @jponge Which legacy reactive streams APIs you means? If I remember correctly, support for Reactive Streams was added in JDK 9.

We are transitioning from `org.reactivestreams.*` to `java.util.concurrent.Flow.*`

**Comment by benkard** [comment]
As it is going to cause some disruption anyway, it seems to me that Quarkus 3 might be a good opportunity to eliminate any remaining [split packages and colliding modules](https://github.com/quarkusio/quarkus/issues/25795) and add explicit `Automatic-Module-Name` declarations to every module.

What would it take to do so?  Is there a way for us in the community to help?

(I realize there are [other issues](https://github.com/quarkusio/quarkus/issues/17747) with Java modules and Quarkus right now, but fixing those probably does not entail any breaking changes, so the timing is less important.)

**Reply by maxandersen** [reply]
You are welcome to make a PR to show what you propose. My current understanding is that making Quarkus JPMS compatible would make the user experience worse for most rather than being a value-add - but I'm happy to be convinced otherwise with code. 

Fixing split packages and colliding modules is useful even outside JPMS thus all for getting those identified and fixed where feasible.

**Comment by mikeri-digg** [comment]
Are there any plans to support distributed cache for hibernate in Quarkus 3? (See https://quarkus.io/guides/hibernate-orm#limitations-of-caching).

**Reply by maxandersen** [reply]
@Sanne @gavinking Can Maybe comment on This.

**Reply by Sanne** [reply]
We won't be able to work on this for the 3.0 release, but it could be something for a subsequent minor release.

**Comment by gesker** [comment]
Will [Jakarta Faces 4.0](https://jakarta.ee/specifications/faces/4.0/)  be part of the incorporation of Jakarta EE 10 APIs and Packages into Quarkus 3.0?

**Reply by maxandersen** [reply]
Making a quarkus-jakarta-faces using a JSF reference implementation seems like an excellent idea for someone into JSF.
We currently recommend those wanting to contribute to making web UI frameworks to do it as an external extension; i.e. in Quarkiverse. Examples of that already existing is quarkus renarde, quinoa, primefaces and myfaces.

**Comment by antoinesd** [comment]
What about CDI Lite build time extension support ? I’m a bit puzzled and worried that it is not mentioned anywhere since this was designed to make Quarkus an official MP implementation…

**Reply by gsmet** [reply]
@antoinesd sorry for the late reply. We have plans for it. Not sure if it will land in 3.0 but that's something that will arrive in Quarkus for sure.
